### PR TITLE
PXP-9024 Expose Arborist /policy endpoint

### PR DESCRIPTION
**Pull request by paulineribeyre**
Jira Ticket: [PXP-9024](https://ctds-planx.atlassian.net/browse/PXP-9024)

### New Features
- Expose Arborist's "/policy" endpoint for users with access

### CODE REVIEW

File: kube/services/revproxy/gen3.nginx.conf/arborist-service.conf [hunk 1]
```diff
@@ -92,6 +92,25 @@ location = /authz/mapping {
     proxy_pass $upstream/auth/mapping;
 }
 
+location = /authz/policy {
+    if ($csrf_check !~ ^ok-\S.+$) {
+        return 403 "failed csrf check";
+    }
+
```
Comment (Avantol13) [inline kube/services/revproxy/gen3.nginx.conf/arborist-service.conf:99]: You could limit this to GET explicitly here if that's the expectation.

File: kube/services/revproxy/gen3.nginx.conf/arborist-service.conf [hunk 2]
```diff
@@ -92,6 +92,25 @@ location = /authz/mapping {
...
+    # authz check to see if the user has access to the policy data
+    set $authz_resource "/services/arborist/policy";
+    set $authz_method "read";
+    set $authz_service "arborist";
+    # be careful - sub-request runs in same context as this request
+    auth_request /gen3-authz;
+
+    set $proxy_service ${arborist_release_name};
+    set $upstream http://${arborist_release_name}-service.$namespace.svc.cluster.local;
```
Comment (Avantol13) [inline kube/services/revproxy/gen3.nginx.conf/arborist-service.conf:108]: Why the rewrite and proxy_redirect? Can you not just proxy_pass?

Comment (paulineribeyre) [reply kube/services/revproxy/gen3.nginx.conf/arborist-service.conf:108]: we need to pass the query parameters too

File: kube/services/revproxy/gen3.nginx.conf/arborist-service.conf (hunk 1 above)
Comment (paulineribeyre) [reply kube/services/revproxy/gen3.nginx.conf/arborist-service.conf:99]: woops, thanks, i'll add that

**Review by Avantol13 — APPROVED** [review]

**Pull request merged**